FFCDemoPaymentService/Messaging/AmqpReceiver.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Amqp;
+using Amqp.Framing;
+using FFCDemoPaymentService.Messaging.Actions;
+
+namespace FFCDemoPaymentService.Messaging
+{
+    public class AmqpReceiver<T>
+    {
+        private readonly IMessageAction<T> action;
+        private ReceiverLink receiver;
+
+        public AmqpReceiver(Session session, string queueName, IMessageAction<T> messageAction, int credit = 5)
```

Comment:
should we default the credit to 1 as that's what we've defaulted to in our config?